### PR TITLE
Slice cleanup

### PR DESCRIPTION
**Pull request by lrusak**
Not ready for primetime yet

- [x] ff/rw looping in the service.slice add-on
- [x] dt-blob.bin needs updating to support both cm1 and cm3 together 
- [x] No more Kodi patches!!!
- [x] minimal linux patches!!!

### CODE REVIEW

**Comment by lrusak** [comment]
updated to include ff/rw looping support.

@MilhouseVH if you get a sec can you possibly look at the python service. I had some issues getting threading working properly and if you have any better ideas please share :smiley_cat:

**Comment by lrusak** [comment]
ping @chewitt for a nice slice icon please! :+1:

**Comment by lrusak** [comment]
I consider this good to go.

The service.slice add-on is now in fact an add-on that would have to be installed from the repo.

I have also created, https://github.com/LibreELEC/slice-firmware
as I needed to add the dts file for the cm3

**Comment by MilhouseVH** [comment]
> The service.slice add-on is now in fact an add-on that would have to be installed from the repo.

Does this mean a Slice image won't include the Slice add-on, which will then need to be installed from the repository by the user? Wouldn't it be better to include the Slice add-on in the Slice image by adding it to virtual/mediacenter/package.mk? Not sure if it would then need to be enabled by default though (probably would) so that might need to be hacked around (or added to addon-manifest.xml, which might work).

**Comment by lrusak** [comment]
@MilhouseVH I'd rather avoid installing add-ons by default due to kodi's trademark rules.

**Comment by lrusak** [comment]
Can we just merge this in the mean time and I'll work on allowing add-ons to be added to the image later?

**Comment by piotrasd** [comment]
you can add to some dummy addon or add to LE settings addons (just only for slice)
lines to addon.xml

	<requires>
		<import addon="service.led_tools" version="0.1"/>>
	</requires>

than will automatical install from repo ;)

File: packages/linux/package.mk [hunk 1]
```diff
@@ -135,16 +135,10 @@ post_patch() {
   for f in $PROJECT_DIR/$PROJECT/config/*-overlay.dts; do
     [ -f "$f" ] && cp -v $f $PKG_BUILD/arch/$TARGET_KERNEL_ARCH/boot/dts/overlays
   done
-  if [ -f $PROJECT_DIR/$PROJECT/config/dt-blob.dts ]; then
-    cp -v $PROJECT_DIR/$PROJECT/config/dt-blob.dts $PKG_BUILD/arch/$TARGET_KERNEL_ARCH/boot/dts
-  fi
   if [ -n "$DEVICE" ]; then
     for f in $PROJECT_DIR/$PROJECT/devices/$DEVICE/config/*-overlay.dts; do
```
Comment (MilhouseVH) [inline packages/linux/package.mk]: When `$PROJECT_DIR/$PROJECT/devices/$DEVICE/config/*-overlay.dts` doesn't exist (and it doesn't exist at the time of writing) then the next line evaluates to false, which leaves the exit code set to 1 and thus the call to `post_patch()` will fail, meaning the kernel is not built.

adding `exit 0` to the end of `post_patch()` will fix the issue, as will adding `|| true` to any `[ -f "$f" ]` test.

I'll PR a fix shortly.

**Pull request merged**